app/views/comments/partials/index/_list.html.erb

```diff
@@ -1,28 +1,40 @@
 <div class="comments-for-post" data-post-id="<%= post.id %>">
-  <% if post.is_comment_locked? %>
-    Comment section has been locked.
-  <% end %>
-  <div class="row notices">
-    <% if post.comments.hidden(CurrentUser.user).count > 0 || (params[:controller] == "comments" && post.comments.count > Comment::RECENT_COUNT) %>
+  <% if !CurrentUser.user.is_moderator? && post.is_comment_disabled? %>
```

Comment:
This whole section seems messy, and it has some redundancies.
Are you sure that you can't restructure it to avoid those?